tests/cpp/nodes/indexing/test_advanced.cpp

```diff
@@ -84,20 +84,40 @@ TEST_CASE("AdvancedIndexingNode") {
 
         graph.emplace_node<ArrayValidationNode>(B_ptr);
 
-        THEN("The resulting array has the size/shape we expect") {
-            CHECK(B_ptr->dynamic());
-            CHECK_THAT(B_ptr->shape(), RangeEquals({-1}));
-            CHECK(B_ptr->size() == Array::DYNAMIC_SIZE);
-            CHECK_THAT(B_ptr->strides(), RangeEquals({sizeof(double)}));
-            CHECK(B_ptr->ndim() == 1);
+        CHECK(B_ptr->min() == 0);
```

Comment:
Just curious, why remove the `THEN`s?

Reply:
I like this style of tests but gosh they take a long time to compile and run because of the combinatorial explosion. So I am trending towards flattening them a bit.